extensions/amp-ad-network-triplelift-impl/0.1/amp-ad-network-triplelift-impl.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2016 Cloudflare. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpA4A} from '../../amp-a4a/0.1/amp-a4a';
+import {base64UrlDecodeToBytes} from '../../../src/utils/base64';
+
+const AMP_SIGNATURE_HEADER = 'X-AmpAdSignature';
+
+const TRIPLELIFT_BASE_URL = 'ib.3lift.com/ttj';
+const TRIPLELIFT_BASE_A4A_URL = 'amp.3lift.com/_a4a/amp/auction';
+
+/**
+ * This is a minimalistic AmpA4A implementation that primarily gets an Ad
+ * through a source URL and extracts the generated signature
+ * from a HTTP header.  This is then given to A4A to validate against
+ * the cloudflare signing key.
+ */
+export class AmpAdNetworkTripleliftImpl extends AmpA4A {
+
+  /**
+   * @param {!Element} element
+   */
+  constructor(element) {
+    super(element);
+  }
+
+  /** @override */
+  isValidElement() {
+    return this.element.hasAttribute('src') && this.isAmpAdElement();
+  }
+
+  /** @override */
+  getSigningServiceNames() {
+    return ['cloudflare'];
+  }
+
+  /** @override */
+  getAdUrl() {
+    return this.element.getAttribute('src').replace(TRIPLELIFT_BASE_URL, TRIPLELIFT_BASE_A4A_URL);
```

Comment:
Do you want to verify that the src attribute has the TRIPLELIFT_BASE_URL to start? I noticed that you're 3p implementation includes validateSrcPrefix('https://ib.3lift.com/'.  Without it, this could allow a callout to ANY XHR CORS end point (e.g. src to doubleclick).  This is supplied by the publisher so what if it includes javascript(sendMeYourCookies())?